### PR TITLE
Add more naunce to the runtime CI agent

### PR DESCRIPTION
**Pull request by gavofyork**
- If there are no changes to runtime code, exit ok; otherwise:
- If spec_version has changed, then:
  - Exit ok if the runtime is rebuilt, fail otherwise.
- If impl_version has changed, then exit ok.
- If neither version has changed, fail.

### CODE REVIEW

**Comment by gavofyork** [comment]
there is no line 26 in the PR? 

"this way right not those pr" ?

**Comment by gabreal** [comment]
> there is no line 26 in the PR?
exactly, it's about the line which exits when there are no changes to the source files. My question was whether every pull request which changes the `spec_version` should be labelled *breaksapi*. If that's the case then we may miss some this way.

**Comment by gabreal** [comment]
So my suggestion would be to put the "no source file changes" behind the labelling of *breaksapi*.

**Comment by gavofyork** [comment]
`breaksapi` should probably be better named `runtimechanged`, but the main thing is that it signals a potential break with the javascript counterparts

**Comment by gavofyork** [comment]
as long as the runtime has changed and it wasn't explicitly marked as an `impl_version`-only bump, then JS may need updating, hence `breaksapi`.

File: scripts/gitlab/check_runtime.sh [hunk 1]
```diff
@@ -32,60 +32,106 @@ fi
...
+	echo
+	echo "# run github-api job for labelling it breaksapi"
+	curl -sS -X POST \
+		-F "token=${CI_JOB_TOKEN}" \
+		-F "ref=master" \
+		-F "variables[BREAKSAPI]=true" \
+		-F "variables[PRNO]=${CI_COMMIT_REF_NAME}" \
+		${GITLAB_API}/projects/${GITHUB_API_PROJECT}/trigger/pipeline
+
+	exit 0
```
Comment (gabreal) [inline scripts/gitlab/check_runtime.sh]: this one would make the next block unnecessary

**Comment by gavofyork** [comment]
@gabreal addressed issue.

**Pull request merged**